### PR TITLE
Adds threads operators support

### PR DESCRIPTION
**Pull request by yurydelendik**
TODO:
- [x] create tests

### CODE REVIEW

**Comment by sunfishcode** [comment]
Looks good! I think we do want to add a preference to disable support for threads. I think it makes sense to put it in the validator so that we handle atomic opcodes the same way as invalid encodings.

File: src/parser.rs [hunk 1]
```diff
@@ -466,6 +467,80 @@ pub enum Operator<'a> {
...
     I64TruncSSatF64,
     I64TruncUSatF64,
+
+    // 0xFE operators
+    // https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md
+    I32Extend8S,
+    I32Extend16S,
+    I64Extend8S,
+    I64Extend16S,
+    I64Extend32S,
```
Comment (sunfishcode) [inline src/parser.rs]: Can you adjust the comment here to make it clear that the Extend operators are not 0xFE operators?

Comment (yurydelendik) [reply src/parser.rs]: Moved them in the section above

**Comment by sunfishcode** [comment]
Looks good to me!

**Pull request merged**